src/ResourceManagement/Compute/Microsoft.Azure.Management.V2.Compute/VirtualMachineScaleSetSkuTypes.cs

```diff
@@ -0,0 +1,248 @@
+
+        /** Static value STANDARD_F1 for VirtualMachineScaleSetSkuTypes. */
+        public static readonly VirtualMachineScaleSetSkuTypes STANDARD_F1 = new VirtualMachineScaleSetSkuTypes("STANDARD_F1", "Standard");
+
+        /** Static value STANDARD_F2 for VirtualMachineScaleSetSkuTypes. */
+        public static readonly VirtualMachineScaleSetSkuTypes STANDARD_F2 = new VirtualMachineScaleSetSkuTypes("STANDARD_F2", "Standard");
+
+        /** Static value STANDARD_F4 for VirtualMachineScaleSetSkuTypes. */
+        public static readonly VirtualMachineScaleSetSkuTypes STANDARD_F4 = new VirtualMachineScaleSetSkuTypes("STANDARD_F4", "Standard");
+
+        /** Static value STANDARD_F8 for VirtualMachineScaleSetSkuTypes. */
+        public static readonly VirtualMachineScaleSetSkuTypes STANDARD_F8 = new VirtualMachineScaleSetSkuTypes("STANDARD_F8", "Standard");
+
+        /** Static value STANDARD_F16 for VirtualMachineScaleSetSkuTypes. */
+        public static readonly VirtualMachineScaleSetSkuTypes STANDARD_F16 = new VirtualMachineScaleSetSkuTypes("STANDARD_F16", "Standard");
+
+        private Sku sku;
+        private string value;
+        /// <summary>
+        /// Creates a custom value for VirtualMachineSizeTypes.
+        /// </summary>
+        /// <param name="skuName">skuName the sku name</param>
+        /// <param name="skuTier">skuTier thr sku tier</param>
+        public  VirtualMachineScaleSetSkuTypes (string skuName, string skuTier) : this(new Sku { Name =skuName, Tier = skuTier })
+        {
+        }
+
+        /// <summary>
+        /// Creates a custom value for VirtualMachineSizeTypes.
+        /// </summary>
+        /// <param name="sku">sku the sku</param>
+        public  VirtualMachineScaleSetSkuTypes (Sku sku)
+        {
+            this.sku = sku;
+            this.value = this.sku.Name;
+            if (this.sku.Tier != null) {
+                this.value = this.value + "_" + this.sku.Tier;
+            }
+        }
+
+        /// <returns>the sku</returns>
+        public Sku Sku
+        {
+            get
+            {
+                return this.sku;
+            }
+        }
+        public override string ToString()
+        {
+            return this.value;
+        }
+
+        public override int GetHashCode()
+        {
+            return this.value.GetHashCode();
+        }
+
+        public override bool Equals (object obj)
+        {
+
+            string value = this.ToString();
+            if (!(obj is VirtualMachineScaleSetSkuTypes)) {
+                return false;
+            }
+
+            if (obj == this) {
+            return true;
+            }
+            VirtualMachineScaleSetSkuTypes rhs = (VirtualMachineScaleSetSkuTypes) obj;
+            if (value == null) {
+                return rhs.value == null;
+            }
+            return value.Equals(rhs.value);
```

Comment:
to Enhance readability compare either ToString() with ToString or .value with .value, right now comparison is between lhs.ToString() and rhs.value (I got implementation of ToString is to return value, this is to enhance maintainability/readability of code)